tests/components/freebox/test_binary_sensor.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for the Freebox sensors."""
+from copy import deepcopy
+from datetime import timedelta
+from unittest.mock import Mock
+
+from homeassistant.components.freebox.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from .const import DATA_STORAGE_GET_RAIDS, MOCK_HOST, MOCK_PORT
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_raid_array_degraded(hass: HomeAssistant, router: Mock) -> None:
+    """Test raid array degraded binary sensor."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: MOCK_HOST, CONF_PORT: MOCK_PORT},
+        unique_id=MOCK_HOST,
+    )
+    entry.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    assert (
+        hass.states.get("binary_sensor.freebox_server_r2_raid_array_0_degraded").state
+        == "off"
+    )
+
+    # Now simulate we degraded
+    DATA_STORAGE_GET_RAIDS_DEGRADED = deepcopy(DATA_STORAGE_GET_RAIDS)
```

Comment:
Please lowercase variable names for local variables.

Reply:
https://github.com/home-assistant/core/pull/95910